apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -336,6 +336,14 @@ type EnvoyListenerConfig struct {
 	// +optional
 	DisableMergeSlashes *bool `json:"disableMergeSlashes,omitempty"`
 
+	// DisableServerHeaderTransformation signifies we will not modify the Server header.
```

Comment:
```suggestion
	// DisableServerHeaderTransformation signifies Envoy will not modify the Server header.
	// If set to false, the Server header will be set to "envoy". If set to true, the Server header
	// returned from the backend will not be modified.
```

I think you can remove the next two lines with the specific Envoy enum fields, we don't need to expose those to the user